pkg/controller/nodemaintenance/nodemaintenance_controller_init.go

```diff
@@ -26,8 +29,11 @@ func Add(mgr manager.Manager) error {
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
 	r := &ReconcileNodeMaintenance{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+
 	err := initDrainer(r, mgr.GetConfig())
-	Handler = r
+	if err == nil {
+		err = r.checkLeaseSupported()
```

Comment:
You're not actually using the result of `checkLeaseSupported()` here.

Reply:
it is used; here we call: func (r *ReconcileNodeMaintenance) checkLeaseSupported() (error) { in node_maintenance_controller.go, this one sets r.isLeaseSupported and there we call the global function func checkLeaseSupported(cs kubernetes.Interface) (bool, error) {

Reply:
changed the name of the static function to checkLeaseSupportedInternal - to avoid confusion.